roles/chart_verifier/tasks/tests.yml

```diff
@@ -90,29 +100,29 @@
         - chart_stat is defined
         - chart_stat.stat.exists
 
-    - name: "Check values_file stat"
+    - name: "Check if chart.values_file exist"
       stat:
-        path: "{{ chart.values_file }}"
-      register: values_stat
+        path: "{{ item }}"
+      loop: "{{ chart.values_file }}"
+      register: file_stats
       when:
         - chart.values_file is defined
 
-    - name: "Chart values file path not found"
+    - name: "Fail if any values file is missing"
       fail:
-        msg: "values_file cannot be found - {{ chart.values_file }}"
+        msg: "File {{ item.item }} does not exist."
+      loop: "{{ file_stats.results }}"
       when:
-        - values_stat is defined
         - chart.values_file is defined
-        - values_stat.stat.exists | bool == false
+        - item.stat.exists == false
 
     - name: "Copy values_file to working directory"
       copy:
-        src: "{{ chart.values_file }}"
+        src: "{{ item }}"
         dest: "{{ work_chart_dir }}/"
+      loop: "{{ chart.values_file }}"
```

Comment:
Is this one looping through a list that could have URLs? If so, would need to take the basename of the URL

Reply:
This block is to process local files. The chart_file location determines if the files are local or remote. If the chart_file is local, the values files must be local too. If the chart file is remote, the files must be remote. That has been clarified in the readme and is the use case exposed by the requester.

Maybe in the next iteration, we can add support for files in remote (uri) and local (copy/stat).

Thanks